pennylane/template.py

```diff
@@ -0,0 +1,312 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+r"""
+QML Templates
+==========
+
+**Module name:** :mod:`pennylane.template`
+
+.. currentmodule:: pennylane.template
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build
+more complex quantum machine learning models.
+
+For example, you can define and call a circuit-centric quantum classifier
+:cite:`schuld2018circuit` on an arbitrary number of wires and with an
+arbitrary number of blocks in the following way:
+
+.. code-block:: python
+
+    import pennylane as qml
+    from pennylane import numpy as np
+
+    num_wires=4
+    num_blocks=3
+    dev = qml.device('default.qubit', wires=num_wires)
+
+    @qml.qnode(dev)
+    def circuit(weights, x=None):
+        qml.BasisState(x, wires=range(num_wires))
+        qml.template.CircuitCentricClassifier(weights, periodic=True, wires=range(num_wires))
+        return qml.expval.PauliZ(0)
+
+    weights=np.random.randn(num_blocks, num_wires, 3)
+    print(circuit(weights, x=np.array(np.random.randint(0, 1, num_wires))))
+
+
+The handy :func:`Interferometer` function can be used to construct arbitrary interferometers in terms of elementary :class:`~.Beamsplitter` and :class:`~.Rotation` operations, by means of the scheme from :cite:`clements2016optimal`, specified either via the unitary transformation on the bosonic operators or in terms of lists of beamsplitter parameters.
+
+The function :func:`CVNeuralNet` implements the continuous variable neural network architecture from :cite:`killoran2018continuous`. Provided with a suitable array of weights, such neural networks can now be easily constructed and trained with PennyLane.
+
+Summary
+^^^^^^^
+
+.. autosummary::
+  CircuitCentricClassifier
+  CircuitCentricClassifierBlock
+  CVNeuralNet
+  CVNeuralNetLayer
+  Interferometer
+  clements
```

Comment:
Capital letter -> Clements?

Reply:
I wanted to use lower case here as is customary for python functions. The other functions in the template module are capitalized only because then they better fit in with the standard PL operations, which are classes. Note that `clements()` is not an "operation" as it cannot be applied to a set of wires.